src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -924,6 +1002,85 @@ private void EvolutionaryTreeNodeSelected(int generation, uint id)
         UpdateSpeciesPreview(world.SpeciesHistoryList[generation][id]);
     }
 
+    private void PatchListMenuIndexChanged(int index)
+    {
+        var patchName = patchListMenu.Popup.GetItemText(index);
+        var selectedPatch = world.GameProperties.GameWorld.Map.Patches.Values
+            .First(p => p.Name.ToString() == patchName);
+
+        // Get current snapshot
+        var patch = new Patch(selectedPatch.Name, 0, selectedPatch.BiomeTemplate, selectedPatch.BiomeType,
+            world.PatchHistoryList[generationDisplayed][selectedPatch.ID])
+        {
+            TimePeriod = selectedPatch.TimePeriod,
+            Depth = { [0] = selectedPatch.Depth[0], [1] = selectedPatch.Depth[1] },
+            Visibility = selectedPatch.Visibility,
+        };
+
+        PatchListMenuUpdate(patch);
+    }
+
+    private void PatchListMenuUpdate(Patch patch)
+    {
+        patchDisplayed = patch;
+        patchListMenu.Text = patch.Name.ToString();
+
+        if (generationDisplayed != world.CurrentGeneration &&
+            world.MicheHistoryList[generationDisplayed].TryGetValue(patch, out Miche? miche))
+        {
+            micheTree.SetMiche(miche);
+        }
+        else
+        {
+            var cache = new SimulationCache(world.WorldSettings);
+
+            var generateMiche = new GenerateMiche(patch, cache, world.WorldSettings);
+            var newMiche = generateMiche.GenerateMicheTree();
+
+            generateMiche.PopulateMiche(newMiche);
+
+            micheTree.SetMiche(newMiche);
+        }
+    }
+
+    private void MicheTreeNodeSelected(int micheHash)
+    {
+        if (!micheTree.MicheByHash.TryGetValue(micheHash, out var micheData))
```

Comment:
Even if they overlap it would just display the incorrect miche, not error here. Still not great, but I don't think it is an issue in the grand scheme of things.